File: 2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        if target not in words:
            return -1
        n=len(words)
        ans=float('inf')
        for i in range(n):
            if words[i]==target:
                d=abs(i-startIndex)
                ans=min(ans,d,n-d)
        return ans

File: 2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.cs
public class Solution {
    public int ClosestTarget(string[] words, string target, int startIndex) {
        if(!words.Contains(target))
            return -1;
        int n=words.Length;
        int ans=int.MaxValue;
        for(int i=0;i<n;i++){
            if(words[i]==target){
                int d = Math.Abs(i-startIndex);
                ans = Math.Min(ans,Math.Min(d,n-d));
            }
        }
        return ans;
    }
}